### DISCUSSION TITLE
What exactly does Sheldon do?

### DISCUSSION THREAD

**Discussion by ProfessorManhattan** [rossmacarthur/sheldon#173] [Q&A]
Hey, so is this a replacement for `antigen` or `oh-my-zsh`? Also, can it mange completions? My unwieldly configuration is stored here: https://github.com/megabyte-labs/install.doctor/blob/master/home/dot_zshrc --- I'd love for something like Sheldon to replace all the various systems I have integrated but I'm not sure if I'm missing what exactly Sheldon does

**Comment by rossmacarthur** [comment]
Sheldon is much more like `antigen` than `oh-my-zsh`.

All Sheldon really does, is the following
- Download and manage local git repos (or remote files or binaries)
- Generate shell source code to load these things using templated shell code.

Each plugin definition in the config file defines how to download the plugin and how to generate the shell code. You can define re-usable installation templates that can be applied to one or more plugins.

Sheldon does not come with much built-in shell code like other plugin managers, (e.g. calling `compinit`). I can't promise that Sheldon will be able to replace everything in your zshrc and I doubt it will reduce much of the shell code you have, but it will allow you to genericize some of the repeated logic and give you a framework for breaking out your code into "plugins".

Some reasons I built Sheldon:
- `oh-my-zsh` comes with a ton of useful stuff but also plenty of stuff I didn't need, for me it wasn't always easy to choose what code I actually wanted (their lib/*.zsh) is rather large. This is especially since I wanted a very lightweight shell experience. 
- Many plugin managers for Zsh plugins require a `<name>.plugin.zsh` file at the root of the repository to be able to install them and I would encounter one that I would have manually install that `antigen` couldn't handle.
- Plugin managers would often have inscrutable special logic to handle repos like `oh-my-zsh` (`antigen` :eyes:).
- Performance: I hate slow tools! I wanted a fast and lightweight plugin manager, I think I was using zplug before which it turns out [is one of the slowest 🤦](https://github.com/rossmacarthur/zsh-plugin-manager-benchmark).
